### PR TITLE
Use `sqlglot` v23.0.1

### PR DESCRIPTION
**Pull request by sundarshankar89**

### CODE REVIEW

File: tests/unit/snow/test_databricks.py [hunk 1]
```diff
@@ -583,14 +583,14 @@ def test_monthname(dialect_context):
     )
     # Test case to validate `monthname` parsing of date column
     validate_source_transpile(
-        """SELECT DATE_FORMAT(TO_DATE('2015-05-01'), 'MMM') AS MONTH""",
+        """SELECT DATE_FORMAT(cast('2015-05-01' as date), 'MMM') AS MONTH""",
```
Comment (nfx) [inline tests/unit/snow/test_databricks.py:586]: nit: i wonder if we can have UPPERCASE for all SQL keywords/functions as the convention

**Review by nfx — APPROVED** [review]
lgtm

File: tests/unit/snow/test_databricks.py (hunk 1 above)
Comment (sundarshankar89) [reply tests/unit/snow/test_databricks.py:587]: conversation converts them to a single case, our validation comparison ignores them and compares

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**